### ISSUE TITLE
[Feat]: Move to /appium/sessions from /sessions for Appium usage

### ISSUE THREAD

**Issue by KazuCocoa** [appium/appium#19856]
To cover appium-inspector's usage. It is not in w3c spec, so it would be nice to have it as a plugin instead of base-driver side.

https://github.com/appium/appium/pull/19851 

(if needed)

[update] removed the grid's comment as just an exmaple.

Users should enable this plugin explicitly, so appium server won't expose these info unexpectedly with the `/status` endpoint call.

**Comment by eglitise** [comment]
Why does extending the `/status` endpoint require a plugin? As long as the server returns the `ready` and `message` fields in the response, it already fits the spec, and adding any other parameters is still valid. I don't see why it can't be included in the base driver.

**Comment by KazuCocoa** [comment]
Maybe it could. I don't know the grid's format so we could include it in the `/status` to behave similar to grid if needed.
The plugin is just let various appium versions can use it, not only latest future appium that use newer version.

**Comment by eglitise** [comment]
The format used by Selenium Grid is not relevant here, I only mentioned it as an example, in the case that it does fit the spec. Any changes to conform to the Grid's format should be done in the Inspector anyway.
As for compatibility with older Appium versions without this `/status` endpoint - the Inspector can simply use the existing to-be-deprecated `/sessions` endpoint, as it already does.

**Comment by KazuCocoa** [comment]
Current our message is just a string as:
https://github.com/appium/appium/blob/7516d4135a1eec37ba298f407254d825961795af/packages/appium/lib/appium.js#L169-L184
So, maybe adding it with a breaking change could be reasonable...?
I don't have strong opinion to add it in the `message` itself.

**Comment by KazuCocoa** [comment]
-> https://github.com/appium/appium/issues/19857

**Comment by jlipps** [comment]
I don't think this should be a plugin. If we want to ensure W3C compatibility, I think it should either be an additional endpoint behind a vendor prefix (`GET /appium/sessions`, with the same behaviour as currently), or an execute method (`appium: getSessions`). The disadvantage of the execute method is that I don't think we have any globally defined execute methods on the appium driver, whereas we do have globally handled routes. So implementing the route is likely easier and potentially less prone to interfering with what drivers are doing.

**Comment by mykola-mokhnach** [comment]
we currently have nothing to be able to run sessionless execute methods as they still use /session_id/execute/sync API under the hood

**Comment by jlipps** [comment]
(this feature is global enough to be put into the core appium code, IMO. and we could hide it behind a security flag so that administrators must opt into making session data available to consumers)

**Comment by mykola-mokhnach** [comment]
also adding a new non-w3c route into the base driver seems like meh... We have recently deprecated most of these

**Comment by KazuCocoa** [comment]
i personally like to keep `/status` minimal so that the server can get many requests without performance concerns etc. So it works like a liveness endpoint for the appium server instance. (just a comment)

**Comment by jlipps** [comment]
@mykola-mokhnach if you have a good idea for how to add execute method handling to the umbrella server, then I'm ok if we want to do that. Maybe it's not hard.

**Comment by KazuCocoa** [comment]
Renamed the title to add `/appium/sessions` endpoint to keep the `/sessions` functionality in w3c era for appium (which should handle non session id stuff.)

No decision, but one potential good one for inspector expansion.

**Comment by jlipps** [comment]
Yeah, could do this now and deprecate `/sessions` so downstream users can start migrating.

**Comment by eglitise** [comment]
Moving the discussion to #20880
